### PR TITLE
Add docs and good defaults for `DistillationTrainer`

### PR DESCRIPTION
**Pull request by cmpatino**
# What does this PR do?

Adds docs and better defaults for the `DistillationTrainer`.

## Before submitting
- [x] This PR fixes a typo or improves the docs (you can dismiss the other checks if that's the case).

## AI writing disclosure

We welcome the use of AI tools to help with contributions. For transparency and to help us improve our review process, please indicate the level of AI involvement in this PR.

- [x] AI-assisted: some parts were suggested or improved by AI, but the PR was written and reviewed by a human.

## Who can review?

Anyone in the community is free to review the PR once the tests have passed. Feel free to tag members/contributors who may be interested in your PR.

---

> [!NOTE]
> **Medium Risk**
> Adds new documentation but also changes multiple `DistillationConfig` defaults (e.g., on-policy fraction, KL mode, top-k, completion length, vLLM memory, learning rate), which can materially alter training behavior for users relying on defaults.
> 
> **Overview**
> Adds a new `DistillationTrainer` documentation page (with quickstart, dataset expectations, and external teacher-server constraints) and links it from the experimental docs toctree.
> 
> Updates `DistillationConfig` to use more opinionated defaults geared toward on-policy distillation and teacher-server usage (e.g., `lmbda=1.0`, `beta=1.0`, `loss_top_k=1`, longer `max_completion_length`, lower `vllm_gpu_memory_utilization`, higher `learning_rate`) and clarifies the `log_completions` help text.
> 
> <sup>Reviewed by [Cursor Bugbot](https://cursor.com/bugbot) for commit 611379fd11cb0443d39736d22ab430aa0ce0c5b3. Bugbot is set up for automated code reviews on this repo. Configure [here](https://www.cursor.com/dashboard/bugbot).</sup>

### CODE REVIEW

**Review by kashif — APPROVED** [review]

File: trl/experimental/distillation/distillation_config.py [hunk 1]
```diff
@@ -136,7 +136,9 @@ class DistillationConfig(_BaseConfig):
         > Parameters that control logging
 
         log_completions (`bool`, *optional*, defaults to `False`):
```
Comment (sergiopaniego) [inline trl/experimental/distillation/distillation_config.py:138]: The `help` in `log_completions: bool = field(...` should contain the same message, no?

**Review by sergiopaniego — APPROVED** [review]
thanks!! For the documentation side, do you feel like that from a user view, the difference between this and GKDTrainer would be clear?

File: trl/experimental/distillation/distillation_config.py
Comment (kashif) [inline trl/experimental/distillation/distillation_config.py]:
```suggestion
        metadata={
            "help": "Whether to log a sample of (prompt, completion) pairs every `log_completions_steps` steps. If `rich` is "
            "installed, it prints the sample. If `wandb` and/or `trackio` logging is enabled, it logs it to `wandb` "
            "and/or `trackio`."
        },
```

File: trl/experimental/distillation/distillation_config.py (hunk 1 above)
Comment (cmpatino) [reply trl/experimental/distillation/distillation_config.py:138]: You're right. Thanks for pointing it out

Comment (cmpatino) [reply trl/experimental/distillation/distillation_config.py:138]: The main differences between this trainer and the `GKDTrainer` are a buffer for better on-policy generation and support for an external teacher server.

In terms of design, it doesn't inherit from `SFTTrainer` as `GKDTrainer` does, and the long-term plan is to upgrade `DistillationTrainer` from experimental status.

I mentioned the `GKDTrainer` explicitly in the docs to make the differences more explicit.
